packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -225,7 +224,8 @@ export class Wallet extends EventEmitter<WalletEvents> {
       channelIds.map(async channelId => {
         const closeResult = await this._engine.closeChannel({channelId});
         const {newObjective, channelResult} = closeResult;
-
+        // create the promise before we send anything out
```

Comment:
Curious, why is it important create a promise here instead of below, in the return dictionary declaration?

Reply:
A few lines lower we call `await this.handleEngineOutput(closeResult)` which will emit events for objectives in `closeResult`.  If `this.handleEngineOutput(closeResult)` is called before we setup the promise then the objective event may be emitted **before** we construct the promise that resolves when the event is emitted.

That means we can end up with a promise that never resolves even though the objective is complete.

Reply:
Thanks for the explanation.